File: Gestao_Utilizadores/Domain/Users/UserDto.cs
using System;

namespace UserManagement.Domain.Users
{
  public class UserDto(Guid id, string email, string firstName, string lastName, string role, bool active, string taxPayerNumber, string mechanographicNumber, string phoneNumber)
  {
    public Guid Id { get; set; } = id;
    public string Email { get; set; } = email;
    public string FirstName { get; set; } = firstName;
    public string LastName { get; set; } = lastName;
    public string Role { get; set; } = role;
    public bool Active { get; set; } = active;
    public string TaxPayerNumber { get; set; } = taxPayerNumber;
    public string MechanographicNumber { get; set; } = mechanographicNumber;
    public string PhoneNumber { get; set; } = phoneNumber;
  }
}

File: Frontend/src/dto/userDTO.ts
export interface UserResponseDTO {
  id: string;
  email: string;
  password: string,
  firstName: string,
  lastName: string;
  role: string;
  active: boolean;
  taxPayerNumber: string;
  mechanographicNumber: string;
  phoneNumber: string;
}

export interface CreateUserRequestDTO {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  role: string;
  active: boolean;
  taxPayerNumber: string;
  mechanographicNumber: string;
  phoneNumber: string;
}
